Lib/tarfile.py

```diff
@@ -1454,14 +1460,14 @@ def _proc_pax(self, tarfile):
         # "%d %s=%s\n" % (length, keyword, value). length is the size
         # of the complete record including the length field itself and
         # the newline. keyword and value are both UTF-8 encoded strings.
-        regex = re.compile(br"(\d+) ([^=]+)=")
+        regex = re.compile(br"^(\d{1,20}) ([^=]+)=")
         pos = 0
-        while match := regex.match(buf, pos):
+        while match := regex.match(buf[pos:]):
```

Comment:
Why is this change? It makes the complexity quadratic, because at every iteration you make a copy of `buf`.

Reply:
It's the opposite. `^` matches at the beginning of the string or after `\n` (if multiline flag is set). If you use `match()`, `^` has no effect if used in `match()` without starting index. So this change does not make sense.

Actually, the code may be simplified/optimized even more by using `finditer()`.

Reply:
The usage of the index as a parameter and splicing the buffer do act different:

```
import re
example_string = "\nabcdefg"
regex = re.compile("^abc")
ind = 1 # Some search index
# See different result:
print(regex.match(example_string, ind))
print(regex.match(example_string[ind:]))
```

the results from the print statements differ despite the regex having a carrot at the start. If the regex is changed to have a carrot at the start, then some tests will fail. With this buffer splicing all tests pass, but it is quite inefficient.